### PR TITLE
Add carbon emissions reporting to job information page

### PR DESCRIPTION
**Pull request by Renni771**
Addresses #15046. This PR adds an initial implementation of carbon emissions reporting to the job information page. It currently assumes global average values for carbon intensity and power usage effectiveness and does not take the user's or server's location into account. The next step would be to dynamically determine the CPU configuration and location of the server that ran a specific job to allow for more accurate estimates.

Screenshots of the new UI:

The new UI using dummy data:
Screenshot from 2023-05-31 11-17-02

Info can be accessed on hover and click:
[Screencast from 2023-05-31 12-05-02.webm](https://github.com/galaxyproject/galaxy/assets/57318507/abedf9b7-36af-47f1-a63b-17ec83699449)

A new route explaining how we make the calculations was added:
[Screencast from 2023-05-31 11-17-57.webm](https://github.com/galaxyproject/galaxy/assets/57318507/adc9a76d-9ddd-4f1c-9934-cca0bdc1ef08)

Notes:
- The graph only shows when there is more than one metric to show. i.e, if we only get cpu info from the job metrics endpoint, only the table shows. The table centers itself in this case to better fill the white space.
- A threshold can be set for each `CarbonEmissionsCard` component - these are the tiles on the left of the table and graph. When the value is below the threshold, we say the value is less than or equal to the threshold so as to avoid tiny decimal numbers or values that don't really make sense for the comparison.

Other changes:
- The AWS estimates is now its own `AwsEstimates` component. The component itself was refactored 
- The AWS estimates and `JobMetrics`  tests were refactored to support the new component splitting.
- The `ec2.json` contained reference data explaining how various AWS EC2 machines were configured. This was used to determine the AWS estimates. This file was updated and converted to a `.js` file with more up-to-date values listed in the AWS EC2 docs. Since the file is large, we dynamically import it on demand and pass its data as props to the `AwsEstimates` component which needs the same data. For now this same EC2 data is used to estimate the configuration of a server instance that could have run the current job. This is the final fallback when we have little or no information from the server about the server the ran the job. 
- An `Abbreviation` component was added.
- The `Heading` component was converted to typescript.
- A `GetComponentPropTypes` utility type was added.

## How to test the changes?
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -43,6 +43,7 @@
     "backbone": "1.4.1",
     "bootstrap": "4.6",
     "bootstrap-vue": "^2.23.0",
+    "chart.js": "4.3.0",
     "citation-js": "^0.6.4",
```
Comment (bgruening) [inline client/package.json]: Can we use d3.js here and avoid this extra dependencie?

Comment (Renni771) [reply client/package.json]: Sure, I've swapped it out for D3.

**Comment by bgruening** [comment]
@Renni771 please feel free to add a screenshot to make it easier for people to review.

**Review by ElectronicBlueberry — APPROVED** [review]
Thanks for all the work on this, looks and works great!

**Comment by Renni771** [comment]
@hexylena sure, pushing the changes just now. A note: since the grid is responsive, there is a range in which the car values won't appear in the same row since we always want to show the total emissions and energy usage first. This can be seen when resizing the inner "analysis" panel.

**Pull request merged**